crates/bevy_ui/src/ui_node.rs

```diff
@@ -1844,22 +1844,69 @@ impl TargetCamera {
     }
 }
 
+#[derive(Component)]
+/// Marker used to identify default cameras, they will have priority over the [`PrimaryWindow`] camera.
+///
+/// This is useful if the [`PrimaryWindow`] has two cameras, one of them used
+/// just for debug purposes and the user wants a way to choose the default [`Camera`]
+/// without having to add a [`TargetCamera`] to the root node.
+///
+/// Another use is when the user wants the Ui to be in another window by default,
+/// all that is needed is to place this component on the camera
+///
+/// ```
+/// # use bevy_ui::prelude::*;
+/// # use bevy_ecs::prelude::Commands;
+/// # use bevy_render::camera::{Camera, RenderTarget};
+/// # use bevy_core_pipeline::prelude::Camera2dBundle;
+/// # use bevy_window::{Window, WindowRef};
+///
+/// fn spawn_camera(mut commands: Commands) {
+///     let another_window = commands.spawn(Window {
+///         title: String::from("Another window"),
+///         ..Default::default()
+///     }).id();
+///     commands.spawn((
+///         Camera2dBundle {
+///             camera: Camera {
+///                 target: RenderTarget::Window(WindowRef::Entity(another_window)),
+///                 ..Default::default()
+///             },
+///             ..Default::default()
+///         },
+///         // We add the Marker here so all Ui will spawn in
+///         // another window if no TargetCamera is specified
+///         UiDefaultCameraMarker
+///     ));
+/// }
+/// ```
+pub struct UiDefaultCameraMarker;
+
 #[derive(SystemParam)]
 pub struct DefaultUiCamera<'w, 's> {
     cameras: Query<'w, 's, (Entity, &'static Camera)>,
+    default_cameras: Query<'w, 's, Entity, (With<Camera>, With<UiDefaultCameraMarker>)>,
     primary_window: Query<'w, 's, Entity, With<PrimaryWindow>>,
 }
 
 impl<'w, 's> DefaultUiCamera<'w, 's> {
     pub fn get(&self) -> Option<Entity> {
-        self.cameras
-            .iter()
-            .filter(|(_, c)| match c.target {
-                RenderTarget::Window(WindowRef::Primary) => true,
-                RenderTarget::Window(WindowRef::Entity(w)) => self.primary_window.get(w).is_ok(),
-                _ => false,
-            })
-            .max_by_key(|(e, c)| (c.order, *e))
-            .map(|(e, _)| e)
+        self.default_cameras.get_single().ok().or_else(|| {
```

Comment:
```suggestion
        self.default_cameras.get_single().or_else(|_| {
```

Reply:
That would need to change the return from `Option<Entity>` to `Result<Entity, SomeError>`, I don't really want to change that thing in specific. DefaultUiCamera is used in ui_layout_tree as an explicit `Option`

Reply:
Ah my bad